### PR TITLE
Add support for repository to TestInfrahub class

### PR DESCRIPTION
**Pull request by dgarros**
This PR break down the `infrahub.testing` directory into 2 parts
- `infrahub-testcontainers` A dedicated python module that contains all the logic related to testcontainers to start Infrahub in Pytest using docker compose. The goal is to publish this package separately to make it easier to import it
- `infrahub_sdk.testing`: The schema part of `infrahub.testing` has moved to the SDK

### CODE REVIEW

File: backend/tests/integration_docker/test_propose_change_repository.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        john_branch = await client.get(kind=TESTING_PERSON, id=john.id, branch=branch.name)
+        john_branch.description.value = "new description"
+        await john_branch.save()
+
+        pc = await client.create(
+            kind=CoreProposedChange, name="pc1", source_branch=branch.name, destination_branch=default_branch
+        )
+        await pc.save()
+
+        # breakpoint()
```
Comment (ajtmccarty) [inline backend/tests/integration_docker/test_propose_change_repository.py:88]: leftover

File: backend/tests/integration_docker/test_propose_change_repository.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        john = client.store.get_by_hfid(key=f"{TESTING_PERSON}__John Doe", raise_when_missing=True)
+
+        john_branch = await client.get(kind=TESTING_PERSON, id=john.id, branch=branch.name)
+        john_branch.description.value = "new description"
+        await john_branch.save()
+
+        pc = await client.create(
+            kind=CoreProposedChange, name="pc1", source_branch=branch.name, destination_branch=default_branch
+        )
+        await pc.save()
```
Comment (ajtmccarty) [inline backend/tests/integration_docker/test_propose_change_repository.py:86]: should this verify that that proposed change is actually created?

**Review by ogenstad — APPROVED** [review]
LGTM, but we might need some way to keep the versions in sync as well. We can to it manually the first time as long as we remember it. Do we need to include the commits from the SDK? We'll need to update that one again very soon. Just want to make sure that we end up with the latest one.

**Review by ajtmccarty — APPROVED** [review]

File: backend/tests/integration_docker/test_propose_change_repository.py (hunk 2 above)
Comment (dgarros) [reply backend/tests/integration_docker/test_propose_change_repository.py:86]: all good points, this file is more of an example to get started then a true functional test at some point

**Comment by dgarros** [comment]
> LGTM, but we might need some way to keep the versions in sync as well. We can to it manually the first time as long as we remember it. Do we need to include the commits from the SDK? We'll need to update that one again very soon. Just want to make sure that we end up with the latest one.

The commit is the one from this morning, we'll need to do a new PR after the release of the SDK

**Pull request merged**